subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmPluginServices.java

```diff
@@ -201,23 +214,31 @@ public File getFile() {
         });
     }
 
-    private Action<ConfigurationInternal> configureDefaultTargetPlatform(boolean alwaysEnabled, TaskProvider<JavaCompile> compileTaskProvider) {
+    private <COMPILE extends AbstractCompile & HasCompileOptions> Action<ConfigurationInternal> configureDefaultTargetPlatform(boolean alwaysEnabled, Set<TaskProvider<COMPILE>> compileTasks) {
         return conf -> {
             JavaPluginConvention javaConvention = project.getConvention().findPlugin(JavaPluginConvention.class);
             if (alwaysEnabled || javaConvention == null || !javaConvention.getAutoTargetJvmDisabled()) {
-                JavaCompile javaCompile = compileTaskProvider.get();
-                int majorVersion;
-                if (javaCompile.getOptions().getRelease().isPresent()) {
-                    majorVersion = javaCompile.getOptions().getRelease().get();
-                } else {
-                    int releaseFlag = getReleaseOption(javaCompile.getOptions().getCompilerArgs());
-                    if (releaseFlag != 0) {
-                        majorVersion = releaseFlag;
+                int majorVersion = 0;
+                for (TaskProvider<COMPILE> compileTaskProvider : compileTasks) {
+                    int candidateMajorVersion;
+                    COMPILE compileTask = compileTaskProvider.get();
+                    if (compileTask.getOptions().getRelease().isPresent()) {
+                        candidateMajorVersion = compileTask.getOptions().getRelease().get();
                     } else {
-                        majorVersion = Integer.parseInt(JavaVersion.toVersion(javaCompile.getTargetCompatibility()).getMajorVersion());
+                        int releaseFlag = getReleaseOption(compileTask.getOptions().getCompilerArgs());
+                        if (releaseFlag != 0) {
+                            candidateMajorVersion = releaseFlag;
+                        } else {
+                            candidateMajorVersion = Integer.parseInt(JavaVersion.toVersion(compileTask.getTargetCompatibility()).getMajorVersion());
+                        }
+                    }
+                    if (candidateMajorVersion > majorVersion) {
+                        majorVersion = candidateMajorVersion;
```

Comment:
majorVersion = max(majorVersion, candidateMajorVersion)

Reply:
If you inline that above, you might be able to get rid of some of the nesting here